### PR TITLE
Fix ProseMirror newline issue in Firefox

### PR DESCRIPTION
**Pull request by kachkaev**
## 🌟 What is the purpose of this PR?

This PR investigates two issues in Firefox and Safari and fixes the Firefox one. They are both related to handling breaks in ProseMirror.

**Firefox**

_cursor ends up before the first character on a new line_

https://user-images.githubusercontent.com/608862/150440263-50718d3e-e48d-4e02-8d46-c17e52219b2f.mov

**Safari**

_cursor lags behind when a new paragraph is inserted_

https://user-images.githubusercontent.com/608862/150440255-8df9d9a8-50f3-485a-abd4-275fc5d81ef5.mov

Interestingly, `prosemirror-view` has fixes exactly for these two issues:
https://github.com/ProseMirror/prosemirror-view/blob/128620ca6e68153f3e51ca13ceab358495767a0b/src/viewdesc.js#L376-L401

The official hack did apply in our setup, unlike in [the official ProseMirror example](https://prosemirror.net/examples/basic/). I found [a Discourse thread that looks quite relevant](https://discuss.prosemirror.net/t/wrong-cursor-placement-with-custom-plugin-and-custom-enter-behaviour-firefox-only/4188), but there is no solution available.

nathggns helped me determine that the problem was inside `packages/hash/frontend/src/blocks/page/ComponentView.tsx` → `ignoreMutation`. This function receives DOM updates and needs to ignore the ones that are triggered by React. Unfortunately, this is not easy by definition.

I changed the function so that it no longer ignores text node creation inside an empty paragraph.

## ⚠️ Known issues

- Safari bug is still there (see screenshots below). We need to un-ignore more mutations but I’m not sure which ones. All my attempts lead to infinite update cycles because I end up un-ignoring React mutations by accident.

- "Select all" + typing several characters places the first character at the end of the new text: `elloh` instead of `hello`. This applies to all browsers and is probably related to `ignoreMutation` too.

- Our ProseMirror does not always handle events correctly, especially when text input is done by Playwright at fast pace. I have increased the number of retries from 1 to 2 to reduce the likelihood of CI failures.

## 🐾 Next steps

Investigate how to change  `ignoreMutation` further so that it returns `false` in more cases.

## 🔍 What does this change?

- Firefox users can try out some basic HASH features

### Does this require a change to the docs?

no

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1201037106070087/f) _(internal)_

## ❓ How to test this?

To run the test locally:

1.  `yarn dev`

1.  `yarn seed-data; npx kill-port 9323; yarn workspace @hashintel/hash-playwright playwright test --workers=1 tests/page-creation.spec.ts --headed`

**Chrome & Firefox after fix** (working)

playwright-chrome

**Firefox before fix** (broken)

playwright-ff

**Safari** (still broken)

playwright-safari

I’ve kept artificial `sleep(100)` statements in the test for now. We should have stable behaviour without delays, but that’s the next step. 100 seconds between key presses is long enough even for humans.

### CODE REVIEW

File: packages/hash/frontend/package.json [hunk 1]
```diff
@@ -45,17 +45,18 @@
     "pluralize": "8.0.0",
     "prismjs": "1.25.0",
     "prosemirror-collab": "1.2.2",
-    "prosemirror-commands": "1.1.12",
+    "prosemirror-commands": "1.2.1",
     "prosemirror-dev-tools": "3.1.0",
-    "prosemirror-dropcursor": "1.3.5",
+    "prosemirror-dropcursor": "1.4.0",
+    "prosemirror-example-setup": "1.1.2",
```
Comment (kachkaev) [inline packages/hash/frontend/package.json:51]: TODO: Remove before merging together with `packages/hash/frontend/src/pages/temp-prosemirror.page.tsx`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -188,7 +188,7 @@ jobs:
           yarn workspace @hashintel/hash-frontend start 2>&1 | tee var/logs/frontend.log & ## ampersand enables background mode
           yarn wait-on --timeout 30000 http://localhost:3000
 
-      - run: yarn playwright install-deps
+      - run: yarn playwright install-deps chrome firefox
```
Comment (kachkaev) [inline .github/workflows/ci.yml:191]: Speeds up CI by skipping Safari download

File: packages/hash/playwright/tests/page-creation.spec.ts [hunk 3]
```diff
@@ -10,6 +10,8 @@ const pageName = `Test page ${pageNameSuffix}`;
 const listOfPagesSelector = 'nav header:has-text("Pages")';
 const pageTitleInputSelector = '[placeholder="A title for the page"]';
 
+const modifierKey = process.platform === "darwin" ? "Meta" : "Control";
```
Comment (kachkaev) [inline packages/hash/playwright/tests/page-creation.spec.ts:13]: **Bonus:** fix + test modifier key. We used to press `Meta+b` etc. before, which only worked on macOS. I spotted this when inspecting Playwright snapshots from CI.

### Ubuntu / Windows before

Screenshot 2022-01-26 at 13 23 17

### Ubuntu / Windows after

Screenshot 2022-01-26 at 13 24 49

**Review by nathggns — APPROVED** [review]

**Comment by nathggns** [comment]
We now have a recording of this flakiness occurring https://app.replay.io/recording/59943358-a173-48d3-ac76-5fc778467eda

**Pull request merged**